rasterio/rio/edit_info.py

```diff
@@ -58,7 +59,10 @@ def transform_handler(ctx, param, value):
         except ValueError:
             pass
         try:
-            retval = guard_transform(value)
+            if tastes_like_gdal(value):
```

Comment:
@geowurster I can't speak to whether or not we want to encourage this behavior, but if we are going to allow GDAL style transforms here, I think you should add a comment indicating this is a special exemption from the deprecations elsewhere, and not simply overlooked.